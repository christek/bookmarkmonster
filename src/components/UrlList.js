// @flow

import React, { useState } from 'react';

import Url from './Url'

import type { Urls, Id } from '../types/todos'

export type Props = {
  todos: Urls,
  onUrlClick: (id: Id) => void,
}

const UrlList = ({ todos, tags, onUrlClick }: Props) => {
  const [count, setCount] = useState(0);

  const handleChange = (value) => {
      console.log('change' + value);
  }

  return (
  <ul>
    {todos.map((todo) => (
      <div>
        <Url key={todo.id} {...todo} />
        <select onChange={e => handleChange(e.target.value)}>
          {tags.map((tag) => (
            <option key={tag.id} value={tag.id}>{tag.text}</option>           
          ))}
        </select>
        <button onClick={() => onUrlClick(todo.id)}>+</button>
      </div>
    ))}
  </ul>
)}

export default UrlList
