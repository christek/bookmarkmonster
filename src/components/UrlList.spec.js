// @flow

import React from 'react';
import { shallow } from 'enzyme';

import UrlList from './UrlList';

const setup = (setupProps = {}) => {
  const defaultProps = {
    todos: [],
    onUrlClick: jest.fn()
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <UrlList todos={props.todos} onUrlClick={props.onUrlClick} />
  );

  return {
    props,
    wrapper
  };
};

describe('UrlList', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('with todos', () => {
    const { wrapper, props } = setup({
      todos: [
        {
          text: 'Test AddUrl',
          id: 0
        }
      ]
    });

    test('renders a list of todos', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });
});
