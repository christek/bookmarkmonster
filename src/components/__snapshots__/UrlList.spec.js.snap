// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlList renders without crashing 1`] = `ShallowWrapper {}`;

exports[`UrlList with todos renders a list of todos 1`] = `ShallowWrapper {}`;
